File: 6 GO/Leetcode-1221-SplitAStringIntoBalancedStrings.go
func balancedStringSplit(s string) int {
    var lseen,rseen,ans int = 0,0,0
    for i := 0; i < len(s); i++ {
        if s[i]=='L'{
            lseen++
        } else{
            rseen++
        }
        if lseen==rseen {ans++}
    }
    return ans;
}

File: 2 CPP/Leetcode-1221-SplitAStringIntoBalancedStrings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int lseen=0,rseen=0,ans=0;
        for(char &c:s){
            if(c=='L'){
                lseen+=1;
            }
            else{
                rseen+=1;
            }
            ans+=(lseen==rseen);
        }
        return ans;
    }
};
